Sage/carbon
# feat(search): add styling for dark theme - FE-3191

Introduces a new `variant` prop which can be either light or dark. Introduces a new styling for when
search is not focused and search has a value.

fixes FE-3191

### Proposed behaviour

New styling for dark theme

<img width="1045" alt="Screenshot 2020-11-27 at 15 36 35" src="https://user-images.githubusercontent.com/56251247/100464952-61f47280-30c6-11eb-9c3c-2c5f94e43d70.png">

<img width="1058" alt="Screenshot 2020-11-27 at 15 36 28" src="https://user-images.githubusercontent.com/56251247/100464958-6587f980-30c6-11eb-896d-31619bdad74d.png">

<img width="1063" alt="Screenshot 2020-11-27 at 15 36 18" src="https://user-images.githubusercontent.com/56251247/100464967-6882ea00-30c6-11eb-817b-b53b3f73ce38.png">

<img width="1044" alt="Screenshot 2020-11-27 at 15 36 08" src="https://user-images.githubusercontent.com/56251247/100464976-6caf0780-30c6-11eb-87d1-414e1a3d8683.png">

### Current behaviour

Search component does not currently have a `variant` prop or support light or dark themes.

### Checklist



### Additional context
N/A

### Testing instructions
- `git checkout FE-3191-search_alt_style`
- `npm start` to start Storybook
- Navigate to Design System -> Search -> Search with Alternative Styling
- Navigate to Design System -> Search - > Search with Alternative Styling and No Button
- Designs should match the Figma designs on FE-3191
- Previous styles should not be affected.
